### PR TITLE
Add reusable action for go version updates

### PR DESCRIPTION
**Pull request by shaynafinocchiaro**
# Description

- PR adds a reusable workflow `Update Go Version Workflow` that will update the go version in Go based repositories. In order to sign commits, I've created a GitHub App called CSM Release and the token is generated using that App. This will sign commits as `release-actions-token[bot]`. See: https://github.com/peter-evans/create-pull-request/blob/main/docs/concepts-guidelines.md#commit-signing
- It adds another workflow `Go Version Update` that calls the reusable workflow.  The workflow can be manually triggered or it can be triggered across multiple repositories at once using the `Trigger Go Version Workflow` workflow.
- It adds a `Trigger Go Version Workflow` workflow that will run this workflow across the repositories specified.

Screenshot 2024-09-27 154430

# Issues
List the issues impacted by this PR:

| Issue ID |
| -------- |
| https://github.com/dell/csm/issues/1490         | 

# Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration

- [ ] [Trigger Job](https://github.com/shaynafinocchiaro/test-common-github-actions/actions/runs/11073710463)
- [ ] [Workflow in action](https://github.com/shaynafinocchiaro/test-common-github-actions/actions/workflows/go-version.yaml)

### CODE REVIEW

File: .github/workflows/go-version-workflow.yaml [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+jobs:
+    go-version:
+      name: Update go version
+      runs-on: ubuntu-latest
+      steps:
+      - name: Check latest go version
+        run: |
+          echo "GO_VERSION_STRIPPED=$(curl -sL https://go.dev/VERSION?m=text | awk 'NR==1{print $1}' | tr -cd '[:digit:].')" >> $GITHUB_ENV
+          echo "GO_VERSION=$(curl -sL https://go.dev/VERSION?m=text | awk 'NR==1{print $1}')" >> $GITHUB_ENV
```
Comment (donatwork) [inline .github/workflows/go-version-workflow.yaml:23]: Will the action fail when these fail?

**Review by donatwork — DISMISSED** [review]
Awesome.

File: .github/workflows/trigger-go-workflow.yaml [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+            "$WF_ORG/karavi-metrics-powerflex",
+            "$WF_ORG/csm-metrics-unity",
+            "$WF_ORG/csm-replication",
+            "$WF_ORG/karavi-resiliency",
+            "$WF_ORG/csm-operator",
+            "$WF_ORG/cert-csi",
+            "$WF_ORG/csm",
+            "$WF_ORG/csm-docs",
+            "$INTERNAL_REPOS",
+          ]
```
Comment (eric-young) [inline .github/workflows/trigger-go-workflow.yaml:69]: I would suggest sorting these in alphabetical order as it makes things easier to maintain

File: .github/workflows/go-version-workflow.yaml (hunk 1 above)
Comment (eric-young) [reply .github/workflows/go-version-workflow.yaml:23]: I agree with Don, I'd suggest you explicitly check the values for these and exit with an error if set incorrectly

**Review by alikdell — DISMISSED** [review]

**Review by eric-young — COMMENTED** [review]
Added a few suggestions

File: README.md [hunk 3]
```diff
@@ -100,6 +100,28 @@ jobs:
...
+The workflow does not accept any parameters and can be used from any repository by creating a workflow that resembles the following
+Note: Workflows that call reusable workflows in the same organization or enterprise can use the inherit keyword to implicitly pass the secrets. See: https://docs.github.com/en/actions/sharing-automations/reusing-workflows#passing-inputs-and-secrets-to-a-reusable-workflow.
+
+```yaml
+name: Go Version Update
+
+on:
+  workflow_dispatch:
+  repository_dispatch:
+    types: [go-update-workflow]
```
Comment (ChristianAtDell) [inline README.md:116]: What will trigger this workflow?

File: .github/workflows/trigger-go-workflow.yaml (hunk 2 above)
Comment (alikdell) [reply .github/workflows/trigger-go-workflow.yaml:69]: Sort on first dir or even inside $WF_ORG

File: .github/workflows/go-version-workflow.yaml (hunk 1 above)
Comment (shaynafinocchiaro) [reply .github/workflows/go-version-workflow.yaml]: Fixed

File: .github/workflows/trigger-go-workflow.yaml (hunk 2 above)
Comment (shaynafinocchiaro) [reply .github/workflows/trigger-go-workflow.yaml:69]: Fixed

**Review by donatwork — APPROVED** [review]

**Review by alikdell — APPROVED** [review]

**Pull request merged**